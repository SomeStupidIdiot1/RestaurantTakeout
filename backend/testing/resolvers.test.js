const { mongoose } = require("../app");
const { createClient } = require("./common/util");
const User = require("../models/User");
const Item = require("../models/Item");
const {
  ADD_ITEM,
  EDIT_ITEM,
  DELETE_ITEM,
  DELETE_ALL_ITEMS,
  CREATE_USER,
  LOGIN,
} = require("./common/mutations");
const { GET_ME } = require("./common/queries");
const {
  getFirstUserExample,
  getSecondUserExample,
  login,
} = require("./common/userAndLogin");
afterAll(() => {
  mongoose.connection.close();
});
beforeEach(async () => {
  await User.deleteMany({});
  await Item.deleteMany({});
});
describe("mutations", () => {
  test("normal login", async () => {
    const { mutate } = createClient();

    const first = await getFirstUserExample(mutate, {
      email: "example@gmail.com",
      password: "this is a bad password",
    });
    const second = await getSecondUserExample(mutate, {
      email: "example123@gmail.com",
      password: "this is a bad password",
    });
    const badToken = await login(mutate, "example@gmail.com", "wrong password");
    const badToken2 = await login(
      mutate,
      "adsf@gmail.com",
      "wradsfong password"
    );
    console.log(first.token);
    expect(first.user).toMatchSnapshot();
    expect(second.user).toMatchSnapshot();
    expect(first.token.errors).not.toBeDefined();
    expect(second.token.errors).not.toBeDefined();
    expect(badToken.errors).toBeDefined();
    expect(badToken2.errors).toBeDefined();
  });
  test("add item", async () => {
    const { query, mutate } = createClient();
    const { token } = await getFirstUserExample(mutate);
    await mutate({
      mutation: ADD_ITEM,
      variables: {
        email,
        password,
      },
      
    });
  });
});
