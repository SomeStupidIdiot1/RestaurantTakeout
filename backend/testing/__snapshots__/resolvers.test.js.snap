// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations normal login 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "address": "123 street",
      "email": "example@gmail.com",
      "facebook": "face",
      "instagram": "insta",
      "phone": "123 3211 4564",
      "restaurantName": "some restaurant name here",
      "twitter": null,
      "youtube": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`mutations normal login 2`] = `
Object {
  "data": Object {
    "createUser": Object {
      "address": "123 street",
      "email": "example123@gmail.com",
      "facebook": "face",
      "instagram": "insta",
      "phone": "123 3211 4564",
      "restaurantName": "some restaurant name here",
      "twitter": null,
      "youtube": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
